Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -371,6 +371,25 @@ Document::Document(JS::Realm& realm, const URL::URL& url, TemporaryDocumentForFr
         .has_legacy_override_built_ins_interface_extended_attribute = true,
     };
 
+    m_cursor_blink_timer = Core::Timer::create_repeating(500, [this] {
+        if (!m_cursor_position)
+            return;
+
+        auto node = m_cursor_position->node();
+        if (!node)
+            return;
+
+        if (auto navigable = this->navigable(); !navigable || !navigable->is_focused())
```

Comment:
When I was going through myself, this part feels a bit awkward - I was trying to think through if there was a better way of expressing this

Reply:
We could probably use a CSS animation for the blinking, and not need this timer. Not sure if that would be better or not. :thinking:

Reply:
Yeah I thought this was pretty goofy too - focus seems like a document property, not navigable. But IMO that warrants a separate PR.